File: tests/fixtures/mini-rust/src/io.rs
use std::fs;
use std::path::Path;

pub fn read_file_to_string(path: &Path) -> std::io::Result<String> {
    fs::read_to_string(path)
}

pub fn write_string_to_file(path: &Path, content: &str) -> std::io::Result<()> {
    fs::write(path, content)
}

pub fn list_files_in_directory(dir: &Path) -> std::io::Result<Vec<std::path::PathBuf>> {
    let mut files = Vec::new();
    for entry in fs::read_dir(dir)? {
        let entry = entry?;
        if entry.file_type()?.is_file() {
            files.push(entry.path());
        }
    }
    Ok(files)
}

pub fn file_exists(path: &Path) -> bool {
    path.exists() && path.is_file()
}

File: tests/fixtures/mini-cpp/io.cpp
#include "io.h"
#include <fstream>
#include <sstream>
#include <filesystem>

std::string read_file_to_string(const std::string& path) {
    std::ifstream file(path);
    if (!file.is_open()) {
        return "";
    }
    std::ostringstream ss;
    ss << file.rdbuf();
    return ss.str();
}

void write_string_to_file(const std::string& path, const std::string& content) {
    std::ofstream file(path);
    file << content;
}

std::vector<std::string> list_files_in_directory(const std::string& dir) {
    std::vector<std::string> files;
    for (const auto& entry : std::filesystem::directory_iterator(dir)) {
        if (entry.is_regular_file()) {
            files.push_back(entry.path().string());
        }
    }
    return files;
}

bool file_exists(const std::string& path) {
    return std::filesystem::exists(path) && std::filesystem::is_regular_file(path);
}